tests/test_types.py

```diff
@@ -2926,6 +2980,15 @@ class Model(BaseModel):
     assert Model.model_validate_json(json.dumps({'foo': str(value)})).foo == result
 
 
+def test_path_validation_constrained():
+    ta = TypeAdapter(Annotated[Path, Field(min_length=9, max_length=20)])
```

Comment:
I think we should be using `TypeAdapter` more internally, it makes the test simpler and removes the model from the core schema, which makes debugging easier.